### PR TITLE
fix(DDS): dds instance support template

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  dds instance support template
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  dds instance support template
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dds/' TESTARGS='-run TestAccDDSV3Instance_withConfiguration'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dds/ -v -run TestAccDDSV3Instance_withConfiguration -timeout 360m -parallel 4
=== RUN   TestAccDDSV3Instance_withConfigurationSharding
=== PAUSE TestAccDDSV3Instance_withConfigurationSharding
=== RUN   TestAccDDSV3Instance_withConfigurationReplicaSet
=== PAUSE TestAccDDSV3Instance_withConfigurationReplicaSet
=== RUN   TestAccDDSV3Instance_withConfigurationSingle
=== PAUSE TestAccDDSV3Instance_withConfigurationSingle
=== CONT  TestAccDDSV3Instance_withConfigurationSharding
=== CONT  TestAccDDSV3Instance_withConfigurationSingle
=== CONT  TestAccDDSV3Instance_withConfigurationReplicaSet
--- PASS: TestAccDDSV3Instance_withConfigurationSingle (630.76s)
--- PASS: TestAccDDSV3Instance_withConfigurationReplicaSet (673.08s)
--- PASS: TestAccDDSV3Instance_withConfigurationSharding (742.08s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dds       742.140s
```

### CODE REVIEW

File: docs/resources/dds_instance.md [hunk 1]
```diff
@@ -155,6 +219,15 @@ The `datastore` block supports:
...
   supports wiredTiger engine, and the Enhanced Edition supports rocksDB engine.
 
+The `configuration` block supports:
+
+* `type` - (Required, String, ForceNew) Specifies the node type. Valid value:
+  + For a Community Edition cluster instance, the value can be **mongos**, **shard** or **config**.
+  + For a Community Edition replica set instance, the value is **replica**.
+  + For a Community Edition single node instance, the value is **single**.
+
+* `id` - (Required, String, ForceNew) Specifies the ID of the template.
```
Comment (deer-hang) [inline docs/resources/dds_instance.md:230]: `Changing this creates a new instance.`

Comment (houpeng80) [reply docs/resources/dds_instance.md:230]: done

File: docs/resources/dds_instance.md [hunk 2]
```diff
@@ -23,6 +27,20 @@ resource "huaweicloud_dds_instance" "instance" {
   security_group_id = "{{ security_group_id }}"
   password          = "Test@123"
   mode              = "Sharding"
+  
+  configuration {
```
Comment (ShiChangkuo) [inline docs/resources/dds_instance.md:31]: the example is a basic usage, the user can ignore `configuration` parameter.

File: docs/resources/dds_instance.md [hunk 3]
```diff
@@ -106,6 +167,9 @@ The following arguments are supported:
 * `mode` - (Required, String, ForceNew) Specifies the mode of the database instance. Changing this creates a new
   instance.
 
+* `configuration` - (Optional, List, ForceNew) Specifies the configuration information. The structure is described below.
```
Comment (ShiChangkuo) [inline docs/resources/dds_instance.md:170]: as the `configuration` can not set from API response, it's should add an note in `Import` section.

File: huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go [hunk 4]
```diff
@@ -262,6 +281,22 @@ func resourceDdsDataStore(d *schema.ResourceData) instances.DataStore {
...
+	logp.Printf("[DEBUG] configurationRaw: %+v", configurationRaw)
+	for i := range configurationRaw {
+		configuration := configurationRaw[i].(map[string]interface{})
+		flavorReq := instances.Configuration{
+			Type: configuration["type"].(string),
+			Id:   configuration["id"].(string),
+		}
+		configurations = append(configurations, flavorReq)
+	}
+	logp.Printf("[DEBUG] flavors: %+v", configurations)
```
Comment (ShiChangkuo) [inline huaweicloud/services/dds/resource_huaweicloud_dds_instance_v3.go:296]:
```suggestion
	logp.Printf("[DEBUG] configurations: %+v", configurations)
```

**Pull request closed without merging**